packages/react-reconciler/src/ReactFiberWorkLoop.new.js

```diff
@@ -1772,24 +1780,31 @@ function resetChildLanes(completedWork: Fiber) {
     let actualDuration = completedWork.actualDuration;
     let treeBaseDuration = ((completedWork.selfBaseDuration: any): number);
 
-    // When a fiber is cloned, its actualDuration is reset to 0. This value will
-    // only be updated if work is done on the fiber (i.e. it doesn't bailout).
-    // When work is done, it should bubble to the parent's actualDuration. If
-    // the fiber has not been cloned though, (meaning no work was done), then
-    // this value will reflect the amount of time spent working on a previous
-    // render. In that case it should not bubble. We determine whether it was
-    // cloned by comparing the child pointer.
-    const shouldBubbleActualDurations =
-      completedWork.alternate === null ||
-      completedWork.child !== completedWork.alternate.child;
-
     let child = completedWork.child;
     while (child !== null) {
       newChildLanes = mergeLanes(
         newChildLanes,
         mergeLanes(child.lanes, child.childLanes),
       );
-      if (shouldBubbleActualDurations) {
+
+      subtreeTag |= child.subtreeTag;
+      subtreeTag |= child.effectTag & HostEffectMask;
+      if (child.deletions.length > 0) {
+        subtreeTag |= Deletion;
```

Comment:
Can we add this tag where we do `child.deletions.push(...)` so we only do this work when we hit a deletion in reconciliation instead of checking this for every Fiber in a render?

Reply:
Maybe. Naively doing this breaks ~150 tests, but it seems like it should be doable. We'll look into it and get back to you.

Reply:
Okay, yup. This is doable too. I'll leave TODO comments to rename the effect tag to something more appropriate once we've merged new->old.